go/mysql/bufio_pool.go

```diff
@@ -0,0 +1,113 @@
+package mysql
```

Comment:
tests please :)

lots of random stuff and make sure we get the right string out the right side.

also i'm curious about error conditions etc